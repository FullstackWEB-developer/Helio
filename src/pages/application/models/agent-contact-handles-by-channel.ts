import {ChannelTypes} from '@shared/models';

export interface AgentContactHandlesByChannel {
    agent: string;
    channel: ChannelTypes;
    startInterval?: Date;
    endInterval?: Date;
    contactsHandled?: number
}
