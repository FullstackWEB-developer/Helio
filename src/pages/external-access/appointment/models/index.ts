export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
