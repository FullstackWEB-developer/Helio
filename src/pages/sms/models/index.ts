export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
