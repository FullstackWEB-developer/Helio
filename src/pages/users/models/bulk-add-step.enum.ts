export enum BulkAddStep {
    Selection = 1,
    RolePicking,
    ProviderMapping,
    Review
}