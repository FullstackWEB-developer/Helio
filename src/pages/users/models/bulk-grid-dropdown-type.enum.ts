export enum BulkGridDropdownType {
    Role,
    ProviderMapping
}