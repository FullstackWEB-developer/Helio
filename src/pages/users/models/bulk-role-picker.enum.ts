export enum BulkRolePicker {
    Group,
    Individual
}