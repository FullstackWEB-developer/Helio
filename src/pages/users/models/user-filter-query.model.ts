export interface UserQueryFilter {
    searchText?: string;
    jobTitle?: string;
    departments?: string;
    roles?: string;
    invitationStatuses?: string;
    statuses?: string;
    rolesUnassigned?: string;
}