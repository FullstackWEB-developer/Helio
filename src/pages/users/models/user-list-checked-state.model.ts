import {CheckboxCheckEvent} from '@components/checkbox/checkbox';
import {UserDetailStatus, UserInvitationStatus} from '@shared/models';

export interface UserListCheckedState {
    checkboxCheckEvent: CheckboxCheckEvent;
    userStatus: UserDetailStatus;
    userInvitationStatus?: UserInvitationStatus;
    userEmail: string;
}