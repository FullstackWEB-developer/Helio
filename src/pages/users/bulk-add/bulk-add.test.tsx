import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {fireEvent, render, wait, waitFor} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import BulkAdd from './bulk-add';
import { act } from 'react-dom/test-utils';
import Router from "react-router";
import Api from '@shared/services/api';
import { ExternalUser, InviteUserModel, PagedList, SelectExternalUser, UserActiveDirectory } from '@shared/models';
jest.setTimeout(30000);
jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useParams: jest.fn().mockReturnValue({ type: "ticket-department" }),
  }))
describe("BulkAdd tests", () => {
    
    let container: HTMLDivElement | null;
    let mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        usersState: {
            selectedExternalUsers: [
                {
                    inviteUserModel: {
                        email: "string0",
                        firstName: "string0",
                        lastName: "string0",
                        name: "string0",
                        providerId: "string0",
                        roles: ["string0"]
                    } as InviteUserModel,
                    id: "string0",
                    info: {
                        azureId: "string0",
                    } as ExternalUser
                }
            ] as SelectExternalUser[],
            bulkPaging: {
                page: 1,
                pageSize: 25,
                totalCount: 2,
                totalPages: 1
            },
            selectedUsersLocalPagination: {
                page: 1,
                pageSize: 25,
                totalCount: 2,
                totalPages: 1
            }
        },
        layoutState: {
            isNavigationExpanded: false
        },
        lookupsState: {
            providerList: [
                {
                    firstName: "string",
                    specialty: "string",
                    homeDepartment: "string",
                    specialtyId: 0,
                    schedulingName: "string",
                    providerTypeId: "string",
                    billable: false,
                    displayName: "string",
                    ansiNameCode: "string",
                    lastName: "string",
                    id: 0,
                    providerUserName: "string",
                    ansiSpecialtyCode: "string",
                    hideInPortal: true,
                    sex: "string",
                    entityType: 0,
                    providerType: "string",
                    createEncounterOnCheckin: false
                },{
                    firstName: "string",
                    specialty: "string",
                    homeDepartment: "string",
                    specialtyId: 1,
                    schedulingName: "string",
                    providerTypeId: "string",
                    billable: false,
                    displayName: "string",
                    ansiNameCode: "string",
                    lastName: "string",
                    id: 1,
                    providerUserName: "string",
                    ansiSpecialtyCode: "string",
                    hideInPortal: true,
                    sex: "string",
                    entityType: 0,
                    providerType: "string",
                    createEncounterOnCheckin: false
                }
            ],
            roleList: [
                {
                    name: "string0",
                    description: "string0",
                },{
                    name: "string1",
                    description: "string1",
                }
            ]
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        snackbarState: {
            messages: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders BulkAdd correctly", async () => {
        let fragment: any;
        await act(async () => {
            const {asFragment} = render(<TestWrapper mockState={mockState}>
                <BulkAdd/>          
            </TestWrapper>);
            fragment = asFragment;
        });
        expect(fragment()).toMatchSnapshot();
    });

    it("renders BulkAdd correctly with data", async () => {
        var response = {
            data: {
                page: 1,
                pageSize: 10,
                totalPages: 1,
                totalCount: 1,
                results: [
                    {
                        azureId: "190981e5-df9a-44b8-b534-da788eba29a3",
                        displayName: "Test",
                        description: "Empty",
                        mail: "test@gmail.com",
                    },{
                        azureId: "190981e5-df9a-44b8-b534-da788eba29b3",
                        displayName: "Test",
                        description: "Empty",
                        mail: "test@gmail.com",
                    }
                ]
            } as PagedList<UserActiveDirectory>
        } as any
        jest.spyOn(Api, 'get').mockResolvedValue(response);
        jest.spyOn(Api, 'post').mockResolvedValue(null);
        await act(async () => {
            const {getByTestId} =render(<TestWrapper mockState={mockState}>
                <BulkAdd/>          
            </TestWrapper>);
            await waitFor(() => {
                fireEvent.click(getByTestId('external-user-0'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('deselect'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('external-user-0'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.change(getByTestId("users.bulk_section.pick_a_role"), {
                    target: {
                        value: "string0"
                    }
                });
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('back'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.input(getByTestId("personal-message"), {
                    target: {
                        value: "string0"
                    }
                });
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await new Promise((r) => setTimeout(r, 5000));
        });
        expect(Api.get).toHaveBeenCalledTimes(7);
        expect(Api.post).toHaveBeenCalledTimes(1);
    });
    it("renders BulkAdd correctly with data error", async () => {
        var response = {
            data: {
                page: 1,
                pageSize: 10,
                totalPages: 1,
                totalCount: 1,
                results: [
                    {
                        azureId: "190981e5-df9a-44b8-b534-da788eba29a3",
                        displayName: "Test",
                        description: "Empty",
                        mail: "test@gmail.com",
                    },{
                        azureId: "190981e5-df9a-44b8-b534-da788eba29b3",
                        displayName: "Test",
                        description: "Empty",
                        mail: "test@gmail.com",
                    }
                ]
            } as PagedList<UserActiveDirectory>
        } as any
        jest.spyOn(Api, 'get').mockResolvedValue(response);
        jest.spyOn(Api, 'post').mockImplementation(() => {
            return Promise.reject({statusCode: 404});
        });
        await act(async () => {
            const {getByTestId} =render(<TestWrapper mockState={mockState}>
                <BulkAdd/>          
            </TestWrapper>);
            await waitFor(() => {
                fireEvent.click(getByTestId('external-user-0'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('deselect'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('external-user-0'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.change(getByTestId("users.bulk_section.pick_a_role"), {
                    target: {
                        value: "string0"
                    }
                });
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('back'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.input(getByTestId("personal-message"), {
                    target: {
                        value: "string0"
                    }
                });
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await new Promise((r) => setTimeout(r, 5000));
        });
        expect(Api.get).toHaveBeenCalledTimes(7);
        expect(Api.post).toHaveBeenCalledTimes(1);
    });

    it("renders BulkAdd correctly with data without roles", async () => {
        var response = {
            data: {
                page: 1,
                pageSize: 1,
                totalPages: 2,
                totalCount: 2,
                results: [
                    {
                        azureId: "190981e5-df9a-44b8-b534-da788eba29a3",
                        displayName: "Test",
                        description: "Empty",
                        mail: "test@gmail.com",
                    },{
                        azureId: "190981e5-df9a-44b8-b534-da788eba29b3",
                        displayName: "Test",
                        description: "Empty",
                        mail: "test@gmail.com",
                    }
                ]
            } as PagedList<UserActiveDirectory>
        } as any
        jest.spyOn(Api, 'get').mockResolvedValue(response);
        mockState.usersState.selectedExternalUsers[0].inviteUserModel.roles = undefined;
        await act(async () => {
            const {getByTestId} =render(<TestWrapper mockState={mockState}>
                <BulkAdd/>          
            </TestWrapper>);
            await waitFor(() => {
                fireEvent.click(getByTestId('external-user-0'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('continue'));
            });
            await waitFor(() => {
                fireEvent.click(getByTestId('back'));
            });
        });
        expect(Api.get).toHaveBeenCalledTimes(7);
    });
})