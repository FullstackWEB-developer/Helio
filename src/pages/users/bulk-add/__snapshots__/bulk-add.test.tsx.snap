// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkAdd tests renders BulkAdd correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-auto h-full"
  >
    <div
      class="hidden"
    >
      <div
        class="pt-4 px-6 flex flex-col"
      >
        <div
          class="flex justify-between h-14 items-center pb-2"
        >
          <div
            class="subtitle"
          >
            users.filters.title
          </div>
          <div
            class="body2 cursor-pointer"
            data-testid="apply"
          >
            users.filters.apply
          </div>
          <div
            class="body2 cursor-pointer"
            data-testid="clear"
          >
            users.filters.clear_all
          </div>
        </div>
        <form>
          <div>
            <div
              class="cursor-pointer flex flex-row justify-between items-center h-12"
            >
              <div
                class="subtitle"
              >
                users.filters.department
              </div>
              <div>
                <div
                  class=""
                >
                  <svg
                    class="cursor-pointer "
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="24"
                      width="24"
                    />
                    <path
                      class="active-item-icon"
                      d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                      transform="translate(19 8) rotate(90)"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <div>
                <div
                  class="h-9"
                >
                  <label
                    class="flex flex-row items-center checkbox-button"
                  >
                    <input
                      checked=""
                      class="checkbox"
                      data-test-id="departments-checkbox-0"
                      data-testid="departments-checkbox-0"
                      id="departments[0]_0"
                      name="departments[0]"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-control"
                    >
                      <div
                        class=""
                      >
                        <svg
                          class="icon-medium "
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="none"
                            height="24"
                            width="24"
                          />
                          <path
                            class="svg-checkbox"
                            d="M16,18H2a2,2,0,0,1-2-2V2A2,2,0,0,1,2,0H16a2,2,0,0,1,2,2V16A2,2,0,0,1,16,18ZM3.41,7.59h0L2,9l5,5,9-9L14.59,3.58,7,11.17,3.41,7.59Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <div
                      class="truncate relative w-full h-full inline-block"
                    >
                      <span
                        class="w-60 w-60"
                      >
                        common.all
                      </span>
                    </div>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div>
            <div
              class="cursor-pointer flex flex-row justify-between items-center h-12"
            >
              <div
                class="subtitle"
              >
                users.filters.job_title
              </div>
              <div>
                <div
                  class=""
                >
                  <svg
                    class="cursor-pointer "
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="24"
                      width="24"
                    />
                    <path
                      class="active-item-icon"
                      d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                      transform="translate(19 8) rotate(90)"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <div>
                <div
                  class="h-9"
                >
                  <label
                    class="flex flex-row items-center checkbox-button"
                  >
                    <input
                      checked=""
                      class="checkbox"
                      data-test-id="titles-checkbox-0"
                      data-testid="titles-checkbox-0"
                      id="titles[0]_0"
                      name="titles[0]"
                      type="checkbox"
                    />
                    <span
                      class="checkbox-control"
                    >
                      <div
                        class=""
                      >
                        <svg
                          class="icon-medium "
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="none"
                            height="24"
                            width="24"
                          />
                          <path
                            class="svg-checkbox"
                            d="M16,18H2a2,2,0,0,1-2-2V2A2,2,0,0,1,2,0H16a2,2,0,0,1,2,2V16A2,2,0,0,1,16,18ZM3.41,7.59h0L2,9l5,5,9-9L14.59,3.58,7,11.17,3.41,7.59Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <div
                      class="truncate relative w-full h-full inline-block"
                    >
                      <span
                        class="w-60 w-60"
                      >
                        common.all
                      </span>
                    </div>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="p-6 w-full flex flex-col flex-grow"
    >
      <h5
        class="pb-4"
      >
        navigation.users_bulk
      </h5>
      <div
        class="w-full flex flex-col relative"
      >
        <span
          class="caption-caps"
        >
          common.step_x_of_x
        </span>
        <div
          class="w-full h-1 stepper my-2"
        >
          <div
            class="h-full step"
            style="width: 25%;"
          />
        </div>
      </div>
      <div
        class="pt-4 flex flex-col bulk-description"
      >
        <h6>
          <span
            class="emphasized pr-2"
          >
             users.bulk_section.step_x
          </span>
          users.bulk_section.step_1_title
        </h6>
        <div
          class="body2 pt-4 w-10/12"
        >
          <div>
            users.bulk_section.step_1_description
          </div>
        </div>
      </div>
      <div
        class="flex justify-end items-baseline pb-3 h-14"
      >
        <div
          class="flex items-baseline mr-24"
        >
          <div
            class="body2 pr-3"
          >
            users.bulk_section.rows_per_page
          </div>
          <div
            class="w-20"
          >
            <div
              class="select-wrapper relative w-full flex flex-col h-20 "
            >
              <div
                class="select relative flex flex-col "
              >
                <input
                  autocomplete="on"
                  autocorrect="on"
                  class="pl-4 pr-8 body2 pt-2.5 h-10 relative truncate select-trigger activated  "
                  spellcheck="true"
                  type="text"
                  value="25"
                />
                <div
                  class="absolute pt-3 right-4 "
                >
                  <div
                    class="flex flex-row items-start"
                  >
                    <div
                      class="cursor-pointer"
                    >
                      <div
                        class=""
                      >
                        <svg
                          class="icon-medium "
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <rect
                            fill="none"
                            height="24"
                            width="24"
                          />
                          <path
                            class="select-arrow-fill"
                            d="M7,14l1.273-1.273L2.545,7,8.273,1.273,7,0,0,7Z"
                            transform="translate(5 16.273) rotate(-90)"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="absolute py-2 options"
                >
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-5"
                        data-testid="select-cell-text-5"
                      >
                        5
                      </div>
                    </div>
                  </div>
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-10"
                        data-testid="select-cell-text-10"
                      >
                        10
                      </div>
                    </div>
                  </div>
                  <div
                    class="w-full select-cell justify-between flex items-center  body2-white   is-selected  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-25"
                        data-testid="select-cell-text-25"
                      >
                        25
                      </div>
                    </div>
                  </div>
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-50"
                        data-testid="select-cell-text-50"
                      >
                        50
                      </div>
                    </div>
                  </div>
                  <div
                    class="w-full select-cell justify-between flex items-center  body2   cursor-pointer  "
                  >
                    <div
                      class="flex flex-col justify-center w-full"
                      data-test-id="select-cell-icon-content"
                    >
                      <div
                        class="flex items-center pl-4"
                        data-test-id="select-cell-text-100"
                        data-testid="select-cell-text-100"
                      >
                        100
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-72"
        >
          <div
            class="flex flex-row items-center"
          >
            <div
              class="pr-8 text-sm text-gray-400"
            >
              <span>
                tickets.pagination
              </span>
            </div>
            <div
              class=""
            >
              <svg
                class="cursor-pointer "
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(0 24) rotate(-90)"
                  width="24"
                />
                <path
                  class="pagination-inactive-icon"
                  d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                  transform="translate(8 5)"
                />
              </svg>
            </div>
            <input
              class="ml-4 mr-4 text-center border rounded-md w-11"
              type="text"
              value="1"
            />
            <div
              class=""
            >
              <svg
                class="cursor-pointer "
                data-testid="next-page"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="24"
                  transform="translate(0 24) rotate(-90)"
                  width="24"
                />
                <path
                  class="pagination-inactive-icon"
                  d="M7,0,8.273,1.273,2.545,7l5.727,5.727L7,14,0,7Z"
                  transform="translate(16.272 19) rotate(180)"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="h-14"
      >
        <div
          class="flex h-14 px-4 border-t border-b"
        >
          <div
            class="md:w-1/6 truncate"
          >
            <div
              class="h-9 pt-4"
            >
              <label
                class="flex flex-row items-center checkbox-button"
              >
                <input
                  class="checkbox"
                  data-testid="bulk-check-all"
                  id="bulk-check-all_"
                  name="bulk-check-all"
                  type="checkbox"
                />
                <span
                  class="checkbox-control"
                >
                  <div
                    class=""
                  >
                    <svg
                      class="icon-medium "
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        fill="none"
                        height="24"
                        width="24"
                      />
                      <path
                        class="svg-checkbox"
                        d="M16,18H2a2,2,0,0,1-2-2V2A2,2,0,0,1,2,0H16a2,2,0,0,1,2,2V16A2,2,0,0,1,16,18ZM3.41,7.59h0L2,9l5,5,9-9L14.59,3.58,7,11.17,3.41,7.59Z"
                      />
                    </svg>
                  </div>
                </span>
                <div
                  class="truncate relative w-full h-full inline-block"
                >
                  <span
                    class="w-60 w-60"
                  >
                    users.bulk_section.x_users_selected_singular
                  </span>
                </div>
              </label>
            </div>
          </div>
          <div
            class="md:w-1/6 border-r border-l"
          >
            <div
              class="relative h-10 w-full search-input-field flex items-center relative h-full w-1/6"
            >
              <div
                class="absolute pl-4"
              >
                <div
                  class=""
                >
                  <svg
                    class="cursor-pointer icon-small "
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      fill="none"
                      height="16"
                      width="16"
                    />
                    <path
                      class="search-icon-fill"
                      d="M12.7,11.23a6.777,6.777,0,0,0,1.4-4.174A7.02,7.02,0,0,0,7.1,0,7.105,7.105,0,0,0,0,7.056a7.105,7.105,0,0,0,7.1,7.056,6.667,6.667,0,0,0,4.2-1.391l3,2.981a.971.971,0,0,0,1.4,0,.957.957,0,0,0,0-1.391Zm-5.6.8A5.022,5.022,0,0,1,2,7.056a5.1,5.1,0,0,1,10.2,0A5.022,5.022,0,0,1,7.1,12.025Z"
                    />
                  </svg>
                </div>
              </div>
              <input
                autocomplete="off"
                class="py-2.5 h-full w-full search-input-field body2 pl-12"
                data-testid="search"
                placeholder="users.search"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="flex items-center pl-6 body2"
          >
            <div
              class="relative"
            >
              <div
                class=""
              >
                <svg
                  class="icon-medium cursor-pointer mr-4 "
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="none"
                    height="24"
                    width="24"
                  />
                  <path
                    class="filter-icon"
                    d="M10,18h4V16H10ZM3,6V8H21V6Zm3,7H18V11H6Z"
                  />
                </svg>
              </div>
            </div>
            common.filters
          </div>
        </div>
      </div>
      <div
        class="w-full overflow-y-auto relative"
      >
        <div
          class="bulk-user-grid col-template-step-1 head-row caption-caps h-12 px-4"
        >
          <div />
          <div
            class="truncate"
          >
            users.bulk_section.name
          </div>
          <div
            class="truncate"
          >
            users.bulk_section.department
          </div>
          <div
            class="truncate"
          >
            users.bulk_section.job_title
          </div>
        </div>
      </div>
      <div
        class="flex pt-7"
      >
        <button
          class="secondary-medium-button mr-8"
          data-testid="deselect"
          type="button"
        >
          <div
            class="flex flex-row justify-center items-center space-x-2"
          >
            <div>
              users.bulk_section.deselect_all
            </div>
          </div>
        </button>
        <button
          class="medium-button"
          data-testid="continue"
          type="button"
        >
          <div
            class="flex flex-row justify-center items-center space-x-2"
          >
            <div>
              common.continue
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
